lib/rucio/db/sqla/migrate_repo/versions/d1189a09c6e0_oauth2_0_and_jwt_feature_support_adding_.py

```diff
@@ -0,0 +1,159 @@
+    if context.get_context().dialect.name in ['oracle']:
+        drop_constraint('IDENTITIES_TYPE_CHK', 'identities', type_='check')
+        create_check_constraint(constraint_name='IDENTITIES_TYPE_CHK',
+                                table_name='identities',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+        drop_constraint('ACCOUNT_MAP_ID_TYPE_CHK', 'account_map', type_='check')
+        create_check_constraint(constraint_name='ACCOUNT_MAP_ID_TYPE_CHK',
+                                table_name='account_map',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+    elif context.get_context().dialect.name == 'mysql' and context.get_context().dialect.server_version_info[0] == 5:
+        create_check_constraint(constraint_name='IDENTITIES_TYPE_CHK',
+                                table_name='identities',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+        create_check_constraint(constraint_name='ACCOUNT_MAP_ID_TYPE_CHK',
+                                table_name='account_map',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+    elif context.get_context().dialect.name == 'mysql' and context.get_context().dialect.server_version_info[0] == 8:
+        execute('ALTER TABLE ' + schema + 'identities DROP CHECK IDENTITIES_TYPE_CHK')  # pylint: disable=no-member
+        create_check_constraint(constraint_name='IDENTITIES_TYPE_CHK',
+                                table_name='identities',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+        execute('ALTER TABLE ' + schema + 'account_map DROP CHECK ACCOUNT_MAP_ID_TYPE_CHK')  # pylint: disable=no-member
+        create_check_constraint(constraint_name='ACCOUNT_MAP_ID_TYPE_CHK',
+                                table_name='account_map',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+    elif context.get_context().dialect.name == 'postgresql':
+
+        drop_constraint('IDENTITIES_TYPE_CHK', 'identities', type_='check')
+        create_check_constraint(constraint_name='IDENTITIES_TYPE_CHK',
+                                table_name='identities',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+        drop_constraint('ACCOUNT_MAP_ID_TYPE_CHK', 'account_map', type_='check')
+        create_check_constraint(constraint_name='ACCOUNT_MAP_ID_TYPE_CHK',
+                                table_name='account_map',
+                                condition="identity_type in ('X509', 'GSS', 'USERPASS', 'SSH', 'SAML')")
+
+    if context.get_context().dialect.name in ['oracle', 'mysql', 'postgresql']:
```

Comment:
i think it would be better if can you merge these in the ones above? flow is a bit complicated and it's better to be explicit, even if it means duplicating code

Reply:
Done.